tools/generator/src/Logger.swift

```diff
@@ -10,33 +10,57 @@ protocol Logger {
 }
 
 /// A `TextOutputStream` that writes to standard error.
-private class StderrOutputStream: TextOutputStream {
+private final class StderrOutputStream: TextOutputStream {
     func write(_ string: String) {
         fputs(string, stderr)
     }
 }
 
+/// A `TextOutputStream` that writes to a string.
+final class StringOutputStream: TextOutputStream {
+    private(set) var output = ""
+
+    func write(_ string: String) {
+        output += string
+    }
+}
+
 /// The logger that is used when not running tests.
-class DefaultLogger: Logger {
+final class DefaultLogger: Logger {
     private var standardError = StderrOutputStream()
+    private var testOutputStream: StringOutputStream?
+
+    func overrideOutputStreamForTests(_ outputStream: StringOutputStream) {
```

Comment:
Instead of this, could you have the initializer take in the output streams (stdout (which would be new) and stderr), with default values set? Having test specific code feels a bit off to me.